appcelerator-archive/amp
# Stack network stage 1

related to #239 

when creating a stack a default network is created having the same name than the stack
- each service of the stack is attached to this network and have an alias name equal to their short Name (the name declared in the stack file)
- each service of the stack has a public name equal to [stackNalme]-[shortName], the name visible on `docker sevice ls` command

if it exists at least one stack service having at least one publish name (PublishSpec.Name != "") then 
- a network [stackName]-public is also created.
- a haproxy service is added in the stack attached to the network [stackName]-public
- each stack service having at least one PublishSpec.Name != "" is  also attached on [stackName]-public network
- each PublishSpec.Name is an entry in the stack Haproxy service to reverse-proxy the service for each http request having a hostname matching the form: [PublishSpec.Name].*
- an entry is added to the main haproxy (the global one, not attached to a stack) in order to reverseproxy any http request matching the form:   *.[stackName].* to the right stack haproxy service.

see #239 (stage 1) description

test:
- stop your amp: sudo ./swarm stop
- make install (after rm -rf vendor and glide install)
- pull all images: ./swarm pull  (you should pull appcelerator/haproxy:1.0.1
- start your amp: sudo ./swarm start
- launch a local amplifier
- create stack : amp stack up test -f api/rpc/stack/test_samples/sample-03.yml

verify:
- the stack is created (amp command doesn't return error, tthe amplifier log shows no error)
- the services test-pinger and test.pinger2 are vislbe using command: docker service ls
- two new networks have been created: test and test-local, using command docker network ls
- the command: `curl www.test.local.axwaytest.net/ping` returns pong

then, delete the stack: amp stack rm -f test
verify:
- the two stack services are not visible anymore using: docker service ls
- the two stack networks have been deleted, using docker network ls

regression test:
- make test

PS: the amp-ui has been updated to be displayed using host name only (ui.local.axwaytest.net or ui.local.appcelerator.io)
